nncf/experimental/post_training_api/engine.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABC
+from abc import abstractmethod
+
+from typing import List
+from typing import TypeVar
+
+ModelType = TypeVar('ModelType')
+TensorType = TypeVar('TensorType')
+
+
+class Engine(ABC):
+    """
+    The basic class aims to provide the interface to infer the model.
+    """
+
+    def set_model(self, model: ModelType) -> None:
+        self.model = model
+
+    @abstractmethod
+    def infer_model(self, input_data: List[TensorType]) -> List[TensorType]:
```

Comment:
Passing a list of input data is quite dangerous in terms of a potential out-of-memory error.